### ISSUE TITLE
Knife --exit-on-error not working

### ISSUE THREAD

**Issue by bradenwright** [chef/chef#5399]
## Description

As far as I can tell knife --exit-on-error is not working
## Chef Version

Chef Development Kit Version: 0.19.1
chef-client version: 12.15.7
delivery version: master (d86679335580be3de22996ef294b20d525889d8d)
berks version: 5.1.0
kitchen version: 1.13.2
## Platform Version

Ubuntu
## Replication Case

Chef Server / 2 Nodes to run knife ssh against

```
knife ssh "recipe:my_recipe AND chef_environment:staging" "sudo chef-client1" --exit-on-error --concurrency 1
c01-test.staging.example.com sudo: chef-client1: command not found
c01.staging.example.com      sudo: chef-client1: command not found
c02-test.staging.example.com   sudo: chef-client1: command not found
c02.staging.example.com      sudo: chef-client1: command not found
```

Seems like this should only run on 1 node and exit, not run on all 4 nodes.
### NOTE: CHEF CLIENT BUGS ONLY

This issue tracker is for the code contained within this repo -- `knife` functionality (not
plugins)
- [Server issues](https://github.com/chef/chef-server/issues/new)
- [ChefDK issues](https://github.com/chef/chef-dk/issues/new)
- Cookbook Issues (see the https://github.com/chef-cookbooks repos or search [Supermarket](https://supermarket.chef.io) or GitHub/Google)

**Comment by bradenwright** [comment]
Any update on this???

**Comment by lamont-granquist** [comment]
looks fixed by #5978
